### PR TITLE
[Exp PyROOT] Add TCollection and TIter pythonisations and tests

### PR DESCRIPTION
**Pull request by etejedor**
This PR mainly adds pythonisations for `TCollection` and its subclasses. It also adds a pythonisation for `TIter`, which is necessary to iterate over `TCollection`s.

The corresponding unit tests for every pythonisation are also proposed.

### CODE REVIEW

**Comment by etejedor** [comment]
@phsft-bot build just on ROOT-performance-centos7-multicore/default, ROOT-fedora29/python3 with flags -Dpyroot_experimental=ON

**Comment by dpiparo** [comment]
@phsft-bot build just on ROOT-performance-centos7-multicore/default, ROOT-fedora29/python3 with flags -Dpyroot_experimental=ON -DCTEST_TEST_EXCLUDE_NONE=ON

**Comment by dpiparo** [comment]
just wanted to try out...

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py [hunk 1]
```diff
@@ -1,17 +1,100 @@
...
+	if not res:
+		raise ValueError('list.remove(x): x not in list')
+
+def _extend_pyz(self, c):
+	# Parameters:
+    # - self: collection
+    # - c: collection to extend self with
+    lenc = c.GetEntries()
+    it = TIter(c)
+    for i in range(lenc):
```
Comment (dpiparo) [inline bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py]:
```suggestion
    for _ in range(lenc):
```

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_titer.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# define here the pythonisations for TIter as immediate, i.e. executed upon
+# import of the ROOT module
+@pythonization(lazy = False)
+def pythonize_titer():
+    klass = cppyy.gbl.TIter
+
+    # Make TIter a Python iterator.
+    # This makes it possible to iterate over TCollections, since Cppyy
+    # injects on them an `__iter__` method that returns a TIter.
+    klass.__next__ = _next_pyz  # Py3
```
Comment (dpiparo) [inline bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_titer.py]: do we need both for both versions or we can make the two cases?

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py [hunk 3]
```diff
@@ -1,17 +1,100 @@
...
 from ROOT import pythonization
+from cppyy.gbl import TIter
 
 from ._generic import add_len
 
+
+# Python-list-like methods
+
+def _remove_pyz(self, o):
+	# Parameters:
```
[the remark this answers is not in the document]
Comment (etejedor) [reply bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py]: Done!

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_titer.py (hunk 2 above)
Comment (etejedor) [reply bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_titer.py]: I think it is quite common to just assign both to the same function, and it does not harm anyway, see e.g. :
https://portingguide.readthedocs.io/en/latest/iterators.html#new-iteration-protocol-next

File: bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py (hunk 1 above)
Comment (etejedor) [reply bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_tcollection.py]: Done!

**Pull request merged**